src/runtime/kernel.jl

```diff
@@ -119,14 +119,16 @@ function ROCKernel(kernel #= ::HostKernel =#; localmem::Int=0)
     group_segment_size = UInt32(group_segment_size + localmem)
     private_segment_size = executable_symbol_kernel_private_segment_size(exec_symbol)
```

Comment:
Is this advisory? E.g. does the kernel run correctly if it get's less than the requested memory?

Reply:
Probably not... If decreased to a too low value, kernel crashes.
I remember it was added baaack when we were dealing with `HSA_STATUS_ERROR_OUT_OF_RESOURCES` queue errors.
But since then there's been a lot of fixes regarding memory usage, so maybe worth removing it.

Reply:
Removed. Everything works fine without it.